rust/arrow/src/array.rs

```diff
@@ -249,8 +263,113 @@ impl<T: ArrowNumericType> PrimitiveArray<T> {
     }
 }
 
+impl<T: ArrowTemporalType + ArrowNumericType> PrimitiveArray<T>
+where
+    i64: std::convert::From<T::Native>,
+{
+    /// Returns value as a chrono `NaiveDateTime`, handling time resolution
+    ///
+    /// If a data type cannot be converted to `NaiveDateTime`, a `None` is returned.
+    /// A valid value is expected, thus the user should first check for validity.
+    /// TODO: extract constants into static variables
+    pub fn value_as_datetime(&self, i: usize) -> Option<NaiveDateTime> {
+        let v = i64::from(self.value(i));
+        match self.data_type() {
+            DataType::Date32(_) => {
+                // convert days into seconds
+                Some(NaiveDateTime::from_timestamp(v * 86_400, 0))
+            }
+            DataType::Date64(_) => Some(NaiveDateTime::from_timestamp(
+                v / 1_000,
```

Comment:
I think the `(v % 1_000) as u32 * 1_000_000` is not needed but either way is fine.

Reply:
I need the nanoseconds, otherwise I lose a bit of accuracy